<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CaveWindow</class>
 <widget class="QWidget" name="CaveWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>680</width>
    <height>520</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CAVE GENERATOR</string>
  </property>
  <property name="styleSheet">
   <string notr="true">  
MainWindow {
    background-color: #f0f0f0;
}

QLabel {
    font-family: 'Segoe UI', Arial, sans-serif;
    font-size: 12px;
    color: #333333;
}

QPushButton {
    font-family: 'Segoe UI', Arial, sans-serif;
    font-size: 12px;
    padding: 5px 10px;
    border: 1px solid #007bff;
    border-radius: 5px;
    background-color: #007bff;
    color: #ffffff;
}

QPushButton:hover {
    background-color: #0056b3;
}

QLineEdit, QSpinBox {
    font-family: 'Segoe UI', Arial, sans-serif;
    font-size: 12px;
    padding: 5px;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}

QGridLayout {
    spacing: 10px;
}

MazeView {
    border: none;
    background-color: white;
}
</string>
  </property>
  <widget class="CaveView" name="caveView">
   <property name="geometry">
    <rect>
     <x>165</x>
     <y>5</y>
     <width>510</width>
     <height>510</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>9</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
  </widget>
  <widget class="QFrame" name="generateCaveQFrame">
   <property name="geometry">
    <rect>
     <x>6</x>
     <y>339</y>
     <width>156</width>
     <height>135</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QFrame#generateCaveQFrame {
    background-color: #ffffff;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QPushButton" name="generateCaveQPushButton">
    <property name="geometry">
     <rect>
      <x>14</x>
      <y>102</y>
      <width>129</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
     </font>
    </property>
    <property name="text">
     <string>GENERATE CAVE</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>24</x>
      <y>65</y>
      <width>106</width>
      <height>32</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="caveGenerateQGridLayout">
     <item row="0" column="0">
      <widget class="QSpinBox" name="rowsQSpinBox">
       <property name="font">
        <font>
         <family>Segoe UI</family>
         <pointsize>-1</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="keyboardTracking">
        <bool>true</bool>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="columnsQSpinBox">
       <property name="font">
        <font>
         <family>Segoe UI</family>
         <pointsize>-1</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="keyboardTracking">
        <bool>true</bool>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="fillProbabilityQLabel">
    <property name="geometry">
     <rect>
      <x>3</x>
      <y>3</y>
      <width>152</width>
      <height>29</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
     </font>
    </property>
    <property name="text">
     <string>― FILL PROBABILITY ―</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QSpinBox" name="fillProbabilityQSpinBox">
    <property name="geometry">
     <rect>
      <x>53</x>
      <y>30</y>
      <width>54</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="singleStep">
     <number>10</number>
    </property>
    <property name="stepType">
     <enum>QAbstractSpinBox::DefaultStepType</enum>
    </property>
    <property name="value">
     <number>50</number>
    </property>
   </widget>
  </widget>
  <widget class="QFrame" name="selectFileQFrame">
   <property name="geometry">
    <rect>
     <x>5</x>
     <y>5</y>
     <width>154</width>
     <height>68</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QFrame {
    background-color: #ffffff;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QLineEdit" name="selectFileQLineEdit">
    <property name="geometry">
     <rect>
      <x>14</x>
      <y>6</y>
      <width>128</width>
      <height>25</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="selectFileQPushButton">
    <property name="geometry">
     <rect>
      <x>27</x>
      <y>39</y>
      <width>100</width>
      <height>22</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>SELECT FILE</string>
    </property>
   </widget>
  </widget>
  <widget class="QFrame" name="findPathQFrame">
   <property name="geometry">
    <rect>
     <x>5</x>
     <y>78</y>
     <width>155</width>
     <height>125</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QFrame#findPathQFrame {
    background-color: #ffffff;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QLabel" name="birthLimitQLabel">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>6</y>
      <width>155</width>
      <height>26</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
     </font>
    </property>
    <property name="text">
     <string>― BIRTH LIMIT―</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="deathLimitQLabel">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>60</y>
      <width>152</width>
      <height>29</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
     </font>
    </property>
    <property name="text">
     <string>― DEATH LIMIT ―</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QSpinBox" name="birthLimitQSpinBox">
    <property name="geometry">
     <rect>
      <x>57</x>
      <y>30</y>
      <width>40</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>7</number>
    </property>
    <property name="value">
     <number>4</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="deathLimitQSpinBox">
    <property name="geometry">
     <rect>
      <x>57</x>
      <y>85</y>
      <width>40</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>7</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
   </widget>
  </widget>
  <widget class="QFrame" name="startStepsQFrame">
   <property name="geometry">
    <rect>
     <x>5</x>
     <y>209</y>
     <width>155</width>
     <height>125</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QFrame#startStepsQFrame {
    background-color: #ffffff;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QLabel" name="millisecondQLabel">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>6</y>
      <width>155</width>
      <height>26</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
     </font>
    </property>
    <property name="text">
     <string>― MILLISECOND ―</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QSpinBox" name="millisecondQSpinBox">
    <property name="geometry">
     <rect>
      <x>41</x>
      <y>30</y>
      <width>75</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
    <property name="singleStep">
     <number>100</number>
    </property>
    <property name="value">
     <number>400</number>
    </property>
   </widget>
   <widget class="QPushButton" name="nextStepQPushButton">
    <property name="geometry">
     <rect>
      <x>13</x>
      <y>66</y>
      <width>129</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
     </font>
    </property>
    <property name="text">
     <string>NEXT STEP</string>
    </property>
   </widget>
   <widget class="QPushButton" name="startQPushButton">
    <property name="geometry">
     <rect>
      <x>13</x>
      <y>96</y>
      <width>129</width>
      <height>24</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
     </font>
    </property>
    <property name="text">
     <string>START</string>
    </property>
   </widget>
  </widget>
  <widget class="QFrame" name="mazeSelectQFrame">
   <property name="geometry">
    <rect>
     <x>6</x>
     <y>478</y>
     <width>156</width>
     <height>37</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QFrame {
    background-color: #ffffff;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QPushButton" name="mazeSelectQPushButton">
    <property name="geometry">
     <rect>
      <x>8</x>
      <y>7</y>
      <width>142</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>-1</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>MAZE</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CaveView</class>
   <extends>QGraphicsView</extends>
   <header>caveview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
